awscodecommit/Dockerfile

```diff
@@ -1,25 +1,31 @@
-FROM golang:alpine AS builder
+FROM golang:1.13-alpine AS builder
+
+ENV GO111MODULE on
+ENV CGO_ENABLED 0
+ENV GOOS linux
+ENV GOARCH amd64
 
-COPY awscodecommit /go/src/awscodecommit
 WORKDIR /go/src/awscodecommit
 
 RUN apk update && apk add --no-cache openssh-client ca-certificates
-RUN GO111MODULE=on go mod download && \
-    CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -a -installsuffix cgo -o /go/bin/awscodecommit
 
-FROM registry.access.redhat.com/ubi8/ubi-minimal
+COPY awscodecommit/go.mod awscodecommit/go.sum /go/src/awscodecommit/
+RUN go mod download
 
-LABEL name="Triggermesh AWS CodeCommit Event Source" \
-      vendor="Triggermesh" \
-      version="v0.1.0" \
-      release="1" \
-      summary="The Triggermesh CodeCommit Source" \
-      description="This is the Triggermesh Knative Event Source for AWS CodeCommit"
+COPY awscodecommit/ /go/src/awscodecommit/
+RUN go install -installsuffix cgo ./...
```

Comment:
Besides readability, is there any reason not to group the `COPY` statements together and then performing a single `RUN` like in most official Docker images?

Reply:
[issue 45](https://github.com/triggermesh/knative-lambda-sources/issues/45)

It is made for build optimization, so when we change code in `main.go` we wont have to wait to install unchanged dependencies 

i guess it is made mostly for local development because on CI it wont change anything 

@odacremolbap am i right ?

Reply:
If it's for build speed that makes sense indeed.